aten/src/ATen/Declarations.cwrap

```diff
@@ -21,10 +21,6 @@
   return: self
   cname: resize
   cpu_half: True
-  before_call:
-    THPUtils_assert(arg_self->storage->flag & TH_STORAGE_RESIZABLE,
-      "calling resize_ on a tensor that has non-resizable storage. Clone it first "
-      "or create a new tensor instead.");
```

Comment:
Looks like it disappeared when we merged variable and tensor :).  We were left with the less-strict version, which just checks that the storage isn't resized (i.e. that the numpy array and our tensor share the same storage).  @colesbury and I thought that was okay, because it follows the usual tensor resize semantics, but I could see an argument the other way (and clearly whoever put in the assert (you?) thought that way).

As an additional note, even if we kept this assert through the tensor/variable merge, we wouldn't uniformly enforce the strict version, because we added a bunch of other functions that reshape things (i.e. as_strided) that don't preserve the check.  So if we wanted the strict check, we have more work to do.

Reply:
`as_strided` is not the same. `TH_STORAGE_RESIZABLE` prevents you from reallocating the memory (since you're not the one managing it), `as_strided` only changes how you view existing memory (this is valid in this case).

Reply:
Right, that was my point.  The assert in question above was strict in that it prevented _any_ call to resize, even calls that didn't reallocate memory.  If we want that strictness (I'm not saying we do), we would have to add it to new functions.